### ISSUE TITLE
Please add the following so man pages work

### ISSUE THREAD

**Issue by lf94** [eraserhd/kak-ansi#17]
I had to match against a different bufname to read man pages.

```
hook -group ansi global BufCreate '/tmp/man\.[a-zA-Z0-9]+' ansi-enable
```

Additionally, it thinks the man page is edited, even when invoking `kak -ro`. Any ideas on how to fix that one? :)

**Comment by eraserhd** [comment]
Hi!  This shouldn't be necessary.  If you are inside Kakoune, you should be able to run `:man ls` and it will create a temporary buffer (kak-ansi overrides Kakoune's `man-impl`).

If you are outside of Kakoune and using it as your pager, `kak-ansi` should automatically detect and process ANSI codes from `*stdin*` and `*stdin-N*` fifo buffers.

If it's not either of these cases, can you tell me how you are invoking `man` and `kak`?

**Comment by lee-at-zoo-corp** [comment]
Hey (from work :) I'm literally doing `export PAGER="kak -ro"` in my `.bashrc`.  Then I invoke via simply `man <thing>`. Sorry for the long wait on the reply!

**Comment by lee-at-zoo-corp** [comment]
@eraserhd any suggestions?

**Comment by eraserhd** [comment]
Hrmm, this is working for me.  I can run this:

```sh
PAGER="/nix/store/s1i3i3ma6radw59zhsvgkwy28z6w02yy-kakoune-2024.05.18/bin/kak -ro" command man ls
```

(`command` overrides my local shell function, and the path overrides my Kakoune wrapper script.)

It's not as pretty because there's no colors, but it does handle overstrike/underline as intended.

What OS are you on?  I have `man` version 2.13.0, whatever that means.

Note that I have a shell function that wraps Kakoune's `man` command, which gives me nice colors:

```sh
      man() {
          if (( $# == 2 )); then
              kak -e "man $2($1)"
              return $?
          else
              kak -e "man $*"
              return $?
          fi
      }
```

**Comment by KoviRobi** [comment]
I have come across this on Android/Termux, which uses mandoc for man pages (which creates a temporary file, and opens the pager with that file as the argument). OP might also be using the same (though not Android/Termux given it's /tmp -- maybe some BSD?)

My personal workaround on Termux is to do `export MANPAGER="kak -e ansi-enable"`

A related issue with this is that when running `ansi-render` it makes the buffer modified so I need to do `q!` to quit -- anything I can do about this? Maybe this is a separate issue because it could happen with other buffers too (it does seem like this plugin strips the escape codes when I write the file back).

**Comment by lf94** [comment]
Yep same deal for me @KoviRobi I forgot to mention I also need to `:q!`.

@eraserhd I'm using Arch Linux

**Comment by eraserhd** [comment]
Using the Arch Linux docker image with the `latest` tag, I've verified that with `man-db` and `man-pages` installed, it pipes the page to $MANPAGER.

With `mandoc` installed, (which provides `man` and conflicts with `man-db`), it _does_ create a temporary file and essentially passes it as a file to edit.

I think `export MANPAGE="kak -e ansi-enable"` is a good work-around here.

The escape codes are deleted from the buffer so that you don't see them, but when piping or using Kakoune's `man` command, this ends up in a scratch buffer so that it doesn't prevent exiting.  I haven't been in the Kakoune chat for a while and I know they were working on a way to conceal text, so I don't know if there's another way around this, but honestly I have this in my shell environment because Kakoune has much better, colored highlighting:

```sh
      man() {
          if (( $# == 2 )); then
              kak -e "man $2($1)"
              return $?
          else
              kak -e "man $*"
              return $?
          fi
      }
```

**Comment by lf94** [comment]
Awesome, thank you @eraserhd , I'm sorry if this was longer lived than necessary heh. We all greatly appreciate your work on kak-ansi and your time!
